### PR TITLE
Feature/interesting facts/create fact validation

### PR DESCRIPTION
**Pull request by MishaPolishchuk**
Add Fluent Validation to create fact logic

### CODE REVIEW

File: Streetcode/Streetcode.BLL/MediatR/MediatrCommandsQueriesValidatorExtension.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using System.Reflection.Metadata.Ecma335;
+
+namespace Streetcode.BLL.MediatR;
+public static class MediatrCommandsQueriesValidatorExtension
+{
+    public static IRuleBuilderOptions<T, string> TitleNameAccessibility<T>(this IRuleBuilder<T, string> ruleBuilder, IRepositoryWrapper repositoryWrapper)
+        => ruleBuilder
+            .MustAsync(async (titleName, cancellationToken) =>
+                !await repositoryWrapper.FactRepository.FindAll(x => x.Title == titleName).AnyAsync(cancellationToken))
+            .WithMessage("Such title name already exist");
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/MediatR/MediatrCommandsQueriesValidatorExtension.cs:13]: Maybe we should use Resource files for error messages ?

**Review by MrVenik — APPROVED** [review]
Well, thats great job!

**Review by SvyatoS1 — APPROVED** [review]

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Fact/Create/CreateFactCommand.cs [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
 using MediatR;
 using Streetcode.BLL.DTO.Streetcode.TextContent.Fact;
-
+using Streetcode.BLL.Interfaces.Validation;
+using Streetcode.DAL.Entities.Feedback;
 namespace Streetcode.BLL.MediatR.Streetcode.Fact.Create;
 
-public sealed record CreateFactCommand(CreateFactRequestDto Request) : IRequest<Result<CreateFactResponseDto>>;
+public sealed record CreateFactCommand(CreateFactRequestDto Request) :
+    /*IRequest<Result<CreateFactResponseDto>>,*/ IValidatableRequest<Result<CreateFactResponseDto>>;
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Fact/Create/CreateFactCommand.cs]: Remove comments ?

File: Streetcode/Streetcode.BLL/Middleware/ValidationExceptionHandler.cs
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/Middleware/ValidationExceptionHandler.cs:1]: Maybe it is related to #78

File: Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 using Microsoft.EntityFrameworkCore;
+using Streetcode.BLL.Middleware;
```
Comment (MrVenik) [inline Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs:2]: ?

**Pull request merged**